edb/server/pgcon/pgcon.pyx

```diff
@@ -124,8 +128,17 @@ async def connect(connargs, dbname):
             lambda: PGConnection(dbname, loop, connargs), addr)
 
     else:
-        _, pgcon = await loop.create_connection(
+        trans, pgcon = await loop.create_connection(
             lambda: PGConnection(dbname, loop, connargs), host=host, port=port)
+        sock = trans.get_extra_info('socket')
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
```

Comment:
Can you also add a comment explaining why we bother with TCP keepalive here?

Reply:
Sure thing!